src/components/Game.tsx

```diff
@@ -0,0 +1,169 @@
+import { useEffect, useReducer, useState } from 'react';
+import { calculateWinner, createArrayAttributes } from '../utilities/';
+import Board from './Board';
+import GameInfo from './GameInfo';
+import Move from './Move';
+import Footer from './Footer';
+
+/**
+ * Type for game state properties
+ */
+type State = {
+  history: GameHistory;
+  xIsNext: boolean;
+  stepNumber: number;
+  counter: number;
+};
+
+type WinnerProps = {
+  winner: string;
+  winningSquares: Array<number>;
+};
+
+/**
+ * Initilization of game state properties
+ * @returns
+ */
+const initialStateProperties: State = {
+  history: createArrayAttributes(9),
+  xIsNext: true,
+  stepNumber: 0,
+  counter: 0,
+};
+
+/**
+ * Inotilization of game winner properties
+ */
+const winnerProps: WinnerProps = { winner: '', winningSquares: [] };
+
+/**
+ * Handle game logic
+ * @param state
+ * @param action
+ * @returns
+ */
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    // Go to next move
+    case 'nextMove': {
+      if (state.stepNumber >= state.counter || state.counter === 1)
+        return { ...state };
+
+      return {
+        ...state,
+        stepNumber: state.stepNumber + 1,
+        xIsNext: !state.xIsNext,
+      };
+    }
+
+    // Go to previous move
+    case 'prevMove':
+      if (state.stepNumber === 0) return { ...state };
+      return {
+        ...state,
+        stepNumber: state.stepNumber - 1,
+        xIsNext: !state.xIsNext,
+        counter: state.counter,
+      };
+
+    // Reset game
+    case 'reset':
+      return initialStateProperties;
+
+    // Handle each board's cell action when clicked
+    case 'handleClick': {
```

Comment:
```suggestion
    case 'addMove': {
```

I don't think the name handleClick really makes sense in this context. In a reducer, given an action (to the game state in this instance) you're mutating the state in a certain way. handleClick feels too low level in the context of game state, though I think the logic can still fit here.